src/GUI/UI/UITransition.cpp

```diff
@@ -196,7 +136,57 @@ void GUIWindow_Travel::Update() {
     }
 }
 
-void GUIWindow_Transition::Update() {
+//----- (00444839) --------------------------------------------------------
+GUIWindow_IndoorEntryExit::GUIWindow_IndoorEntryExit(HouseId transitionHouse, unsigned exit_pic_id, Vec3f pos, int yaw,
+                                                     int pitch, int zspeed, std::string_view locationName)
+    : GUIWindow_Transition(WINDOW_IndoorEntryExit, SCREEN_INPUT_BLV) {
+    std::string hint;
+
+    engine->_teleportPoint.setTeleportTarget(pos, yaw, pitch, zspeed);
+    engine->_teleportPoint.setTeleportMap(locationName);
+    _transitionStringId = std::to_underlying(transitionHouse); // TODO(Nik-RE-dev): is this correct?
```

Comment:
Btw that's some fishy stuff.

`pTransitionStrings` is accessed both using mapid and houseid, and then we also have `getSpecialTransferMessageIndex`. O_O.

Reply:
MapId is for house transitions which is MM6 stuff. Maybe there they constructed this array so mapids and houseids does not intersect in MM6.
In MM7 this array is indexed by house id only. Also by house id indexed animation array - so when transfer UI is opened we have video of this indoor location.
I looked in event scripts now - for these location listed in `specialTransferMessageLocationNames` transfer is initiated with house id = HOUSE_INVALID. So that's why they needed special array for it that is constructed in the way that index of he map name == index in `pTransitionStrings`. It is also results that no animation is played for such transfers
I cannot fathom why NWC devs needed such elaborate uglyness. My only thought is that animations was not ready or dropped and they programmed these exceptions in such sloppy way.